cmake/CMakeLists.txt

```diff
@@ -118,29 +117,37 @@ if(COLVARS_TCL)
   else()
     target_compile_options(colvars PRIVATE -DCOLVARS_TCL)
   endif()
-  target_link_libraries(colvars PRIVATE ${TCL_LIBRARY})
+  target_link_libraries(colvars PUBLIC ${TCL_LIBRARY})
 endif()
 
 option(COLVARS_TORCH "Link against the Torch library" OFF)
 
 if(COLVARS_TORCH)
-  find_package(Torch)
-  if (Torch_FOUND)
-    set_property(TARGET colvars PROPERTY CXX_STANDARD 17)
-    target_compile_definitions(colvars PRIVATE -DTORCH)
-    # target_compile_options(colvars PRIVATE ${TORCH_CXX_FLAGS})
-    target_compile_options(colvars PRIVATE ${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS})
-    target_include_directories(colvars PRIVATE ${TORCH_INCLUDE_DIRS})
-    target_link_libraries(colvars PRIVATE "${TORCH_LIBRARIES}")
-  else()
-    message(FATAL_ERROR "Torch package not found.")
+  message(STATUS "Trying to find the Torch library from its Python package")
+  find_package(Python3)
+  if(Python3_FOUND)
+    execute_process(
+      COMMAND ${Python3_EXECUTABLE} -c "import sys; print(next(p for p in sys.path if 'site-packages' in p))"
+      OUTPUT_VARIABLE SITE_PACKAGES_PATH
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    find_package(Torch PATHS file(JOIN ${SITE_PACKAGES_PATH} "torch/share/cmake/Torch"))
```

Comment:
A simple `find_package(Torch)` works for me. This doesn't. I usually add `-DCMAKE_PREFIX_PATH=/home/hanatok/HDD/software/libtorch_2.2.2` (I downloaded and unzipped the libtorch library to the path) to compile a library with libtorch.

Reply:
> A simple `find_package(Torch)` works for me. This doesn't. 

There was probably some trickery involved in packaging libTorch for PyPI that I can't figure out.

> I usually add `-DCMAKE_PREFIX_PATH=/home/hanatok/HDD/software/libtorch_2.2.2` (I downloaded and unzipped the libtorch library to the path) to compile a library with libtorch.

The "pip" option was the recommended one at: https://pytorch.org/get-started/locally/

Does the "LibTorch" package include Python modules as well? Better if it did, because in the future I would like to also use your awesome scripts to test gradients. If it doesn't we can treat it in the same way as Charm++ I guess and put it under /opt as an optional dependency.

In any case, I would like a standard scriptable procedure to bundle libTorch in the containers. Manually bundling is the sort of thing that "works on my computer" but not great for cloud CI jobs :frowning_face:

Reply:
> The "pip" option was the recommended one at: https://pytorch.org/get-started/locally/

No. When I select "C++/Java" in Language, the "Package" selection turns to "LibTorch" automatically, so I don't think it recommends "pip". If you select "pip" then you download the python package instead of the C++ package. What we need is the C++ package "libtorch".

Reply:
> In any case, I would like a standard scriptable procedure to bundle libTorch in the containers. Manually bundling is the sort of thing that "works on my computer" but not great for cloud CI jobs ☹️

If you select "LibTorch", "C++/Java" and then "CPU", the website will give you the download link of libtorch. Is it possible to just use the link with wget or curl to download the libtorch for CI?

Reply:
Right, so by this you are confirming that "libTorch" only has the underlying C++ library with no Python bindings. (Or are you using that same library in scripts?)

Reply:
Yes. I believe #570 only needs the libtorch and does not need the python bindings.

Reply:
I updated the container to use the C++ only libTorch. However, using just `-DCMAKE_PREFIX_PATH=/opt/libtorch` *did not* work. I still had to manually feed the prefix to `find_package()`, so I put the detection of it in the CMake script used for tests.

This PR should be working now, at least minus the stuff that was still broken in `master` on the last merge of `torchann`.

Reply:
You can also fill `Torch_DIR` cmake variable with the path to the cmake configfile directory (`-DTorch_DIR=/opt/libtorch/share/cmake/Torch`) instead of `CMAKE_PREFIX_PATH`